### PR TITLE
Log ops size on disconnect

### PR DESCRIPTION
**Pull request by NicholasCouri**
For more information about how to contribute to this repo, visit [this page](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md).

For specific guidelines for Pull Requests in this repo, visit [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

Today when we reach the 1Mb socket.io limit, we have no idea around the size of the ops we tried to submit. This is an 
Attempt to log the ops size when the event _disconnected gets emitted.

### CODE REVIEW

File: packages/test/test-service-load/src/optionsMatrix.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ const gcOptionsMatrix: OptionsMatrix<IGCRuntimeOptions> = {
 };
 
 const summaryOptionsMatrix: OptionsMatrix<ISummaryRuntimeOptions> = {
-    disableIsolatedChannels: booleanCases,
+    disableIsolatedChannels: [undefined],
```
Comment (vladsud) [inline packages/test/test-service-load/src/optionsMatrix.ts:61]: This looks unrelated and should be removed from this PR, right?

**Review by vladsud — APPROVED** [review]

File: packages/loader/container-loader/src/deltaManager.ts [hunk 2]
```diff
@@ -337,6 +344,8 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
         // state. As requirements change, so should these checks.
         assert(this.messageBuffer.length === 0, 0x0e9 /* "messageBuffer is not empty on new connection" */);
 
+        this.opsSize = 0;
```
Comment (andre4i) [inline packages/loader/container-loader/src/deltaManager.ts:347]: If this resets at connect, If I understand correctly, this may report a lot more op sizes, right? Basically everything in-between connects. If we end up in a disconnect loop, the first trace will probably be bad

File: packages/test/test-service-load/src/optionsMatrix.ts (hunk 1 above)
Comment (NicholasCouri) [reply packages/test/test-service-load/src/optionsMatrix.ts]: yes, sorry, forgot to merge.

File: packages/loader/container-loader/src/deltaManager.ts (hunk 2 above)
Comment (NicholasCouri) [reply packages/loader/container-loader/src/deltaManager.ts:347]: that is actually a good point - I wonder if I should also reset when we flush

Comment (andre4i) [reply packages/loader/container-loader/src/deltaManager.ts:347]: Tough question :) Was looking through the code and you're right, we may lose the size.

Comment (vladsud) [reply packages/loader/container-loader/src/deltaManager.ts:347]: I'm not sure what's the concern. There should be no ops sent while client is disconnected (this is enforced). And same ops maybe attempted to be sent on multiple connections - DM has no way to differentiate. I believe that's totally fine, if someone wants to capture how many unique ops were actually sent, they will need to add different type of telemetry (measure acked ops).

Comment (NicholasCouri) [reply packages/loader/container-loader/src/deltaManager.ts:347]: That makes sense.

Comment (andre4i) [reply packages/loader/container-loader/src/deltaManager.ts:347]: My concern was as following:

- delta manager gets created, opsSize is 0
- successful payloads are being transmitted and opsSize is incremented with their lenghts.
- one large 1MB payload comes along, opSize += 1 MB
- disconnect happens, connectionProps will be logged, with opSize == (whatever was before the large payload + the large payload)
- opSize is 0 again
- the large payload is sent again, opSize == 1 MB
- disconnect happens, connectionProps will be logged, with opSize == the actual large payload

Not a big deal, but just saying that the first log statement will contain the large payload size + whatever was sent successfully before that. The subsequent log statements will be correct..
